test/e2e/scheduling/nvidia-gpus.go

```diff
@@ -190,3 +193,103 @@ var _ = SIGDescribe("[Feature:GPUDevicePlugin]", func() {
 		testNvidiaGPUs(f)
 	})
 })
+
+func testNvidiaGPUsJob(f *framework.Framework) {
+	_ = SetupNVIDIAGPUNode(f, false)
+	// Job set to have 5 completions with parallelism of 1 to ensure that it lasts long enough to experience the node recreation
+	completions := int32(5)
+	ginkgo.By("Starting GPU job")
+	StartJob(f, completions)
+
+	job, err := jobutil.GetJob(f.ClientSet, f.Namespace.Name, "cuda-add")
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	// make sure job is running by waiting for its first pod to start running
+	err = jobutil.WaitForAllJobPodsRunning(f.ClientSet, f.Namespace.Name, job.Name, 1)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	numNodes, err := framework.NumberOfRegisteredNodes(f.ClientSet)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	nodes, err := framework.CheckNodesReady(f.ClientSet, numNodes, framework.NodeReadyInitialTimeout)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	ginkgo.By("Recreating nodes")
+	gce.RecreateNodes(f.ClientSet, nodes)
+	ginkgo.By("Done recreating nodes")
+
+	ginkgo.By("Waiting for gpu job to finish")
+	err = jobutil.WaitForJobFinish(f.ClientSet, f.Namespace.Name, job.Name)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	ginkgo.By("Done with gpu job")
+
+	gomega.Expect(job.Status.Failed).To(gomega.BeZero(), "Job pods failed during node recreation: %v", job.Status.Failed)
```

Comment:
Why don't we check this `job.Status.Failed` at line 206?
`job.Status.Failed` doesn't seem to be updated after line 204, am I missing something?

Reply:
There is the node recreation that happens in between line 204 and here. We want to ensure that when nodes are recreated during the job, it doesn't fail.